src/core/main.c

```diff
@@ -3192,6 +3192,11 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        /* If we got a SoftRebootStart timestamp during deserialization, then we are in a new soft-reboot
+         * iteration, so bump the counter now before starting units, so that they can reliably read it. */
+        if (dual_timestamp_is_set(&m->timestamps[MANAGER_TIMESTAMP_SOFTREBOOT_START]))
```

Comment:
hmm, i'd prefer if we had a clean way to detect how we came to be. i.e. maybe add a proper serialization field for this, that encodes ManagerObjective during serialization.

I am pretty sure this is interesting for the new instance in many other ways sooner or later to know if we come back from reexec, from switch root or from soft reboot.

I am not a fan of detecting cases like this as side-effect of some other value, I'd rather have this a primary concept

Reply:
i.e. I'd even add a proper, second ManagerObject field to the Manager struct that tells us what we are deserializing from. SHould be _MANAGER_OBJECTIVE_INVALID if we are spawned for the very first time.